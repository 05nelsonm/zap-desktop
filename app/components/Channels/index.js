export ChannelCapacity from './ChannelCapacity'
export ChannelMoreButton from './ChannelMoreButton'
export ChannelStatus from './ChannelStatus'
export ChannelSummaryList from './ChannelSummaryList'
export ChannelSummaryListItem from './ChannelSummaryListItem'
