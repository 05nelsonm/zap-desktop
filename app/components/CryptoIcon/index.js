import CryptoIcon from './CryptoIcon'

export default CryptoIcon
