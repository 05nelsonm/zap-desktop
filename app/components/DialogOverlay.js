import styled from 'styled-components'
import { Flex } from 'rebass'

const Overlay = styled(Flex)`
  position: absolute;
  z-index: 99999;
  background-color: rgba(255, 255, 255, 0.25);
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`

export default Overlay
