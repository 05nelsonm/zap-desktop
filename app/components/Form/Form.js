import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { MdClose } from 'react-icons/lib/md'

import PayForm from './PayForm'
import RequestForm from './RequestForm'

import styles from './Form.scss'

const FORM_TYPES = {
  PAY_FORM: PayForm,
  REQUEST_FORM: RequestForm  
}

const Form = ({ formType, formProps, closeForm }) => {
  if (!formType) { return null }

  const FormComponent = FORM_TYPES[formType]
  return (
    <div className={`${styles.outtercontainer} ${formType && styles.open}`}>
      <div className={styles.innercontainer}>
        <div className={styles.esc} onClick={closeForm}>
          <MdClose />
        </div>
        
        <div className={styles.content}>
          <FormComponent {...formProps} />
        </div>
      </div>
    </div>
  )
}


Form.propTypes = {
  
}

export default Form
