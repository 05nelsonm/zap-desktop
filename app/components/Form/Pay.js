import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Isvg from 'react-inlinesvg'
import paperPlane from 'icons/paper_plane.svg'

import { FaBolt, FaChain, FaAngleDown } from 'react-icons/lib/fa'
import LoadingBolt from 'components/LoadingBolt'
import CurrencyIcon from 'components/CurrencyIcon'

import styles from './Pay.scss'

class Pay extends Component {
  componentDidUpdate(prevProps) {
    const {
      isOnchain, isLn, payform: { payInput }, fetchInvoice
    } = this.props

    // If on-chain, focus on amount to let user know it's editable
    if (isOnchain) { this.amountInput.focus() }

    // If LN go retrieve invoice details
    if ((prevProps.payform.payInput !== payInput) && isLn) {
      fetchInvoice(payInput)
    }
  }

  render() {
    const {
      payform: { amount, payInput, showErrors },
      currency,
      crypto,

      isOnchain,
      isLn,
      currentAmount,
      usdAmount,
      inputCaption,
      showPayLoadingScreen,
      payFormIsValid: { errors, isValid },

      setPayAmount,
      onPayAmountBlur,

      setPayInput,
      onPayInputBlur,

      onPaySubmit
    } = this.props

    console.log('usdAmount: ', usdAmount)

    return (
      <div className={styles.container}>
        {showPayLoadingScreen && <LoadingBolt />}
        <header className={styles.header}>
          <Isvg src={paperPlane} />
          <h1>Make Payment</h1>
        </header>

        <div className={styles.content}>
          <section className={styles.destination}>
            <div className={styles.top}>
              <label htmlFor='destination'>Destination</label>
              <span>
              </span>
            </div>
            <div className={styles.bottom}>
              <textarea
                type='text'
                placeholder='Payment request or bitcoin address'
                value={payInput}
                onChange={event => setPayInput(event.target.value)}
                onBlur={onPayInputBlur}
                id='destination'
                rows='4'
              />
            </div>
          </section>

          <section className={styles.amount}>
            <div className={styles.top}>
              <label>Amount</label>
              <span></span>
            </div>
            <div className={styles.bottom}>
              <input
                type='number'
                min='0'
                ref={(input) => { this.amountInput = input }}
                size=''
                placeholder='0.00000000'
                value={currentAmount || ''}
                onChange={event => setPayAmount(event.target.value)}
                onBlur={onPayAmountBlur}
                id='amount'
                readOnly={isLn}
              />
              <div className={styles.currency}>
                <section className={styles.currentCurrency}>
                  <span>BTC</span><span><FaAngleDown /></span>
                </section>
                <ul>
                  <li>Bits</li>
                  <li>Satoshis</li>
                </ul>
              </div>
            </div>

            <div className={styles.usdAmount}>
              {`â‰ˆ ${usdAmount} USD`}
            </div>
          </section>

          <section className={styles.submit}>
            <div className={`${styles.button} ${isValid && styles.active}`} onClick={onPaySubmit}>Pay</div>
          </section>
        </div>
      </div>
    )
  }
}


Pay.propTypes = {
  payform: PropTypes.shape({
    amount: PropTypes.string.isRequired,
    payInput: PropTypes.string.isRequired,
    showErrors: PropTypes.object.isRequired
  }).isRequired,
  currency: PropTypes.string.isRequired,
  crypto: PropTypes.string.isRequired,

  isOnchain: PropTypes.bool.isRequired,
  isLn: PropTypes.bool.isRequired,
  currentAmount: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  inputCaption: PropTypes.string.isRequired,
  showPayLoadingScreen: PropTypes.bool.isRequired,
  payFormIsValid: PropTypes.shape({
    errors: PropTypes.object,
    isValid: PropTypes.bool
  }).isRequired,

  setPayAmount: PropTypes.func.isRequired,
  onPayAmountBlur: PropTypes.func.isRequired,
  setPayInput: PropTypes.func.isRequired,
  onPayInputBlur: PropTypes.func.isRequired,
  fetchInvoice: PropTypes.func.isRequired,

  onPaySubmit: PropTypes.func.isRequired
}

export default Pay
