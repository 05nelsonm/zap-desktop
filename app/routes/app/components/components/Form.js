// @flow
import React, { Component } from 'react'
import { FaDollar, FaBitcoin } from 'react-icons/lib/fa'
import { MdArrowBack, MdClose } from 'react-icons/lib/md'
import styles from './Form.scss'

class Form extends Component {
  componentWillMount() {
    this.props.fetchPeers()
  }

  render() {
    const { 
      setAmount,
      setMessage,
      setPubkey,
      payment: { amount, message, pubkey },
      peers: { peers },
      ticker: { currency },
      isOpen,
      close
    } = this.props

    return (
      <div className={`${styles.formContainer} ${isOpen ? styles.open : ''}`}>
        <div className={styles.container}>
          <div className={styles.esc} onClick={close}>
            <MdClose />
          </div>
          <div className={styles.content}>
            <section className={styles.amountContainer}>
              <label>
                {
                  currency === 'btc' ?
                    <FaBitcoin />
                  :
                    <FaDollar />
                }
              </label>
              <input
                type='text'
                size=''
                style={{ width: `${(amount.length * 20) + 10}%`, fontSize: `${190 - (amount.length ** 2)}px` }}
                value={amount}
                onChange={(e) => setAmount(e.target.value)}
              />
            </section>
            <section className={styles.inputContainer}>
              <label>For:</label>
              <input
                type='text'
                placeholder='Dinner, Rent, etc'
                value={message}
                onChange={(e) => setMessage(e.target.value)}
              />
            </section>
            <section className={styles.inputContainer}>
              <label>To:</label>
              <input
                type='text'
                placeholder='Public key'
                value={pubkey}
                onChange={(e) => setPubkey(e.target.value)}
              />
            </section>
            <section className={styles.peersContainer}>
              {
                peers.length ? 
                  <ul className={styles.peers}>
                    <h4>Connected Peers</h4>
                    {
                      peers.map(peer => {
                        return(
                          <li key={peer.pub_key} className={styles.peer} onClick={() => setPubkey(peer.pub_key)}>
                            <p className={styles.address}>{peer.address}</p>
                            <p className={styles.pubkey}>{peer.pub_key}</p>
                            <MdArrowBack />
                          </li>
                        )
                      })
                    }
                  </ul>
                :
                  null
              }
            </section>
            <section className={styles.buttonGroup}>
              <div className={styles.button}>Pay</div>
              <div className={styles.button}>Request</div>
            </section>
          </div>
        </div>
      </div>
    )
  }
}

Form.propTypes = {}

export default Form