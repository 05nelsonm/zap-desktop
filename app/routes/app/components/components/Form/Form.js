import React from 'react'
import PropTypes from 'prop-types'
import { FaDollar, FaBitcoin } from 'react-icons/lib/fa'
import { MdClose } from 'react-icons/lib/md'
import { btc } from '../../../../../utils'
import styles from './Form.scss'

const Form = ({
  form: { formType, amount, message, payment_request },
  setAmount,
  setMessage,
  setPaymentRequest,
  ticker: { currency, btcTicker },
  isOpen,
  close,
  createInvoice,
  payInvoice,
  fetchInvoice,
  formInvoice
}) => {
  const requestClicked = () => {
    createInvoice(amount, message, currency, btcTicker.price_usd)
      .then((success) => {
        if (success) { close() }
      })
  }

  const payClicked = () => {
    payInvoice(payment_request)
      .then((success) => {
        if (success) { close() }
      })
  }

  const paymentRequestOnChange = (payreq) => {
    setPaymentRequest(payreq)
    if (payreq.length === 124) { fetchInvoice(payreq) }
  }

  const calculateAmount = value => (currency === 'btc' ? btc.satoshisToBtc(value) : btc.satoshisToUsd(value, btcTicker.price_usd))

  return (
    <div className={`${styles.formContainer} ${isOpen ? styles.open : ''}`}>
      <div className={styles.container}>
        <div className={styles.esc} onClick={close}>
          <MdClose />
        </div>
        <div className={styles.content}>
          <section className={styles.amountContainer}>
            <label>
              {
                currency === 'btc' ?
                    <FaBitcoin />
                  :
                    <FaDollar />
              }
            </label>
            <input
              type='text'
              size=''
              style={
                formType === 'pay' ?
                  { width: '75%', fontSize: '100px' }
                :
                  { width: `${amount.length > 1 ? (amount.length * 15) - 5 : 25}%`, fontSize: `${190 - (amount.length ** 2)}px` }
              }
              value={formType === 'pay' ? calculateAmount(formInvoice.amount) : amount}
              onChange={event => setAmount(event.target.value)}
              readOnly={formType === 'pay'}
            />
          </section>
          {
            formType === 'pay' ?
              <section className={styles.inputContainer}>
                <label>Request:</label>
                <input
                  type='text'
                  placeholder='Payment Request'
                  value={payment_request}
                  onChange={(event) => paymentRequestOnChange(event.target.value)}
                />
              </section>
            :
              <section className={styles.inputContainer}>
                <label>For:</label>
                <input
                  type='text'
                  placeholder='Dinner, Rent, etc'
                  value={message}
                  onChange={(e) => setMessage(e.target.value)}
                />
              </section>
          }
          {
            formType === 'pay' ? 
              <section className={styles.buttonGroup}>
                <div className={styles.button} onClick={payClicked}>
                  Pay
                </div>
              </section>
            :
              <section className={styles.buttonGroup}>
                <div  className={styles.button} onClick={requestClicked}>
                  Request
                </div>
              </section>
          }
        </div>
      </div>
    </div>
  )
}

Form.propTypes = {
  form: PropTypes.object.isRequired,
  ticker: PropTypes.object.isRequired,
  formType: PropTypes.string, 
  amount: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  message: PropTypes.string,
  payment_request: PropTypes.string,
  setAmount: PropTypes.func.isRequired,
  setMessage: PropTypes.func.isRequired,
  setPaymentRequest: PropTypes.func.isRequired,
  currency: PropTypes.string,
  btcTicker: PropTypes.object,
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  createInvoice: PropTypes.func.isRequired,
  payInvoice: PropTypes.func.isRequired,
  fetchInvoice: PropTypes.func.isRequired,
  formInvoice: PropTypes.object.isRequired
}

export default Form
