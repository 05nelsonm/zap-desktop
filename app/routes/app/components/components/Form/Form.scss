@import '../../../../../variables.scss';

.formContainer {
	position: absolute;
	top: 0;
	bottom: 0;
	width: 100%;
	height: 100vh;
	background: $white;
	z-index: 0;
	opacity: 0;
	transition: all 0.5s;

	&.open {
		opacity: 1;
		z-index: 10;
	}
}

.container {
	position: relative;
	height: 100vh;
	margin: 5%;
}

.esc {
	position: absolute;
	top: 0;
	right: 0;
	color: $darkestgrey;
	cursor: pointer;
  padding: 20px;
	border-radius: 50%;

	&:hover {
		color: $bluegrey;
    background: $darkgrey;
	}

	&:active {
		color: $white;
		background: $main;
	}

	svg {
		width: 32px;
		height: 32px;
	}
}

.content {
	width: 50%;
	margin: 0 auto;
	display: flex;
	flex-direction: column;
	height: 75vh;
	justify-content: center;
	align-items: center;

	.amountContainer {
  	color: $main;
  	display: flex;
  	justify-content: center;
    min-height: 120px;
    margin-bottom: 20px;
		
		label, input[type=text] {
			color: inherit;
			display: inline-block;
			vertical-align: top;
			padding: 0;
		}

		label {
			svg {
				width: 100px;
				height: 100px;
			}	
		}
		
		input[type=text] {
    	width: 100px;
			font-size: 180px;
			border: none;
    	outline: 0;
    	-webkit-appearance: none;
		}
	}

	.inputContainer {
    width: 100%;
		display: flex;
		justify-content: center;
		font-size: 18px;
		height: auto;
    min-height: 55px;
    margin-bottom: 20px;
    border: 1px solid $traditionalgrey;
    border-radius: 6px;
    position: relative;
    padding: 0 20px;

		label, input[type=text] {
			font-size: inherit;
		}

		label {
			padding-top: 19px;
    	padding-bottom: 12px;
    	color: $traditionalgrey;
		}

		input[type=text] {
			width: 100%;
			border: none;
    	outline: 0;
    	-webkit-appearance: none;
    	height: 55px;
    	padding: 0 10px;
		}
	}

	.peersContainer {
		width: 100%;
    margin: 10px 0 50px 0;

		h4 {
			text-transform: uppercase;
			color: $black;
			letter-spacing: 2;
			font-weight: bold;
			margin-bottom: 20px;
			padding: 10px 0;
		}

		.peers {
			overflow-y: scroll;
			width: 100%;
		}

		.peer {
			position: relative;
			width: 100%;
			border-top: 1px solid $traditionalgrey;
			border-right: 1px solid transparent;
			border-bottom: 1px solid transparent;
			border-left: 1px solid transparent;
			padding: 5px;
			cursor: pointer;
			transition: all 0.25s;

			&:hover {
				background: lighten($main, 20%);
				border-radius: 6px;
				border: 1px solid $main;

				svg {
					visibility: visible;
				}
			}

			p {
				margin: 3px 0;
			}

			.address {
				font-size: 18px;
			}

			.pubkey {
				font-size: 12px;
			}

			svg {
				visibility: hidden;
				position: absolute;
				top: calc(50% - 8px);
				right: 5%;
				color: $black;
			}
		}
	}

	.buttonGroup {
		width: 100%;
		display: flex;
		flex-direction: row;
		border-radius: 6px;
    overflow: hidden;

    .button {
    	cursor: pointer;
			height: 55px;
			min-height: 55px;
			text-transform: none;
			font-size: 18px;
			transition: opacity .2s ease-out;
			background: $main;
	    color: $white;
	    border: none;
	    font-weight: 500;
	    padding: 0;
	    width: 100%;
	    text-align: center;
	    line-height: 55px;

	    &:first-child {
	    	border-right: 1px solid lighten($main, 20%);
	    }
    }
	}
}
