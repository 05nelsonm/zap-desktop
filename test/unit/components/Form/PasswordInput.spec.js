import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { PasswordInput } from 'components/Form'

describe('component.UI.PasswordInput', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<PasswordInput field="password" />)
    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
