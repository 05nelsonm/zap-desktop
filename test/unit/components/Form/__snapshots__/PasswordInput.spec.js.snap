// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.PasswordInput should render correctly 1`] = `
<PasswordInput___StyledInput
  field="password"
  type="password"
>
  <PasswordInput__StyledEyeIcon
    onClick={[Function]}
  />
</PasswordInput___StyledInput>
`;
